src/credentials/providers/sharedCredentialsProviderFactory.ts

```diff
@@ -0,0 +1,89 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as fs from 'fs-extra'
+import { getLogger, Logger } from '../../shared/logger'
+import {
+    getConfigFilename,
+    getCredentialsFilename,
+    loadSharedCredentialsProfiles,
+    updateAwsSdkLoadConfigEnvironmentVariable
+} from '../sharedCredentials'
+import { BaseCredentialsProviderFactory } from './credentialsProviderFactory'
+import { SharedCredentialsProvider } from './sharedCredentialsProvider'
+
+export class SharedCredentialsProviderFactory extends BaseCredentialsProviderFactory<SharedCredentialsProvider> {
```

Comment:
This isn't really a factory, it's more like a collection, maybe `SharedCredentialsProvider` would be a better name?

Reply:
SharedCredentialsProviderFactory produces SharedCredentialsProvider objects (around line 62)

Reply:
Right but it doesn't produce them to be used elsewhere and continues to manage the lifecycle of them so it's not really a factory

Reply:
What do you think about replacing factory with repository?

Reply:
Repository sounds more appropriate